internal/app/machined/pkg/controllers/network/dns_upstream.go

```diff
@@ -103,36 +103,22 @@ func (ctrl *DNSUpstreamController) run(ctx context.Context, r controller.Runtime
 		return err
 	}
 
-	for i, s := range rs.TypedSpec().DNSServers {
-		remoteAddr := s.String()
+	initConn, err := existingConnections(ctx, r)
+	if err != nil {
+		return err
+	}
+
+	for i, srv := range rs.TypedSpec().DNSServers {
+		remoteHost := srv.String()
 
 		if err = safe.WriterModify[*network.DNSUpstream](
 			ctx,
 			r,
-			network.NewDNSUpstream(remoteAddr),
+			network.NewDNSUpstream("№"+strconv.Itoa(i)+" "+remoteHost),
```

Comment:
"№" wha's this :thinking:

Reply:
It's [numero sign](https://en.wikipedia.org/wiki/Numero_sign). I always forget that it's mostly replaced by # now.